src/psyclone/psyGen.py

```diff
@@ -1914,6 +1928,49 @@ def _encloses_omp_directive(self):
             #                       "any OpenMP directives. This is probably "
             #                       "not what you want.")
 
+    def update(self):
+        ''' 
+        Updates the fparser2 AST by inserting nodes for this OpenMP
+        parallel region.
+
+        :raises InternalError: if the existing AST doesn't have the \
+                               correct structure to permit the insertion \
+                               of the OpenMP parallel region.
+        '''
+        from fparser.common.readfortran import FortranStringReader
+        from fparser.two.Fortran2003 import Comment
+        # Check that we haven't already been called
+        if self._ast:
+            return
+        # Find the locations in which we must insert the begin/end
+        # directives...
+        # Find the children of this node in the AST of our parent node
+        try:
+            start_idx = self._parent._ast.content.index(self._children[0]._ast)
+            end_idx = self._parent._ast.content.index(self._children[-1]._ast)
+        except:
+            raise InternalError("Failed to find locations to insert "
+                                "begin/end directives.")
+        # Create the start directive
+        text = "!$omp parallel default(shared), private({0})".format(
+            ",".join(self._get_private_list()))
+        startdir = Comment(FortranStringReader(text,
+                                               ignore_comments=False))
+
+        # Create the end directive and insert it after the node in
+        # the AST representing our last child
+        enddir = Comment(FortranStringReader("!$omp end parallel",
+                                             ignore_comments=False))
+        if end_idx == len(self._parent._ast.content) - 1:
```

Comment:
It's unclear to me why you have this if clause here

Reply:
Because I assumed that `some_list.insert(len(some_list), my_obj)` would be an error but it isn't - I've removed the if clause.